docs/docs/FAQ/Machine-Learning-FAQ.md

```diff
@@ -0,0 +1,61 @@
+---
+sidebar_position: 6
+---
+# Machine Learning
+
+### How does smart search work?
+
+Immich uses CLIP models, for more information about CLIP and its capabilities can be [read here](https://openai.com/research/clip).
+
+### How does facial recognition work?
```

Comment:
I believe the question you're actually answering is "Which models are being used?"

Reply:
This was actually my suggestion. I was thinking we could flesh it out some more and have that be the question. Do you think it'd be better for them to be separate questions?

Reply:
I basically don't know anything about machine learning stuff and reading the respective text did not explain to me how facial recognition works. I don't think I have a preference on either making it two questions or extending the current text. All I'm saying is that the question right now isn't being answered by the text :)

Reply:
Actually I can try to expand on this, but if we're going to expand on it, maybe you should put it in one of these:

https://immich.app/docs/features/facial-recognition/
or here
https://immich.app/docs/administration/jobs
or
https://immich.app/docs/developer/architecture

In any case, I think that an extended explanation would definitely be something welcome, it would allow contributors to the code or people to understand how the models work in Immich. I understand that it may not be of interest to everyone, but people can be directed to read a research article if they are interested in the depth of things

Here is my suggestion:

Immich uses InsightFace to perform
face detection and recognition.
Every time a new property is uploaded to Immich, thumbnails are created for it.
InsightFace uses machine learning to identify faces in thumbnails, for each face a unique identifier is created, also called an embed, and it belongs to the Postgers database under the face_asset column where the embed values of all the faces in the image + the position of the face in the image are saved, this is saved as a vector value, Immich performs a comparison of :

The embed created for the face from thumbnails of the uploaded asset
Compared to the embeds that exist in a database

Before the additional association there are 2 values that Immich is set to check according to:

1. If a value greater than the MIN DETECTION SCORE defined by the user is received, then Immich will not consider it as a face to add

2. If a value equal to or higher than the MAX RECOGNITION DISTANCE defined by the user is received, then Immich will insert the embed found in the database into the embed of an existing person (if any) if there is no existing person, Immich will create a new person.

For each new embed, Immich performs a test of the 100 **closest** embeds to the new embed, so that there is no unnecessary load on the system for searching among all the embeds in the database.


@mertalev  What do you think of this explanation?
I hope the translation was able to keep my explanation clear enough.